minidump-processor/src/stackwalker/amd64.rs

```diff
@@ -281,13 +281,13 @@ where
             // It's unclear why, perhaps the test is buggy, but for now we
             // preserve that behaviour.
             if let Some(last_bp) = last_bp {
-                let address_of_bp = address_of_ip - POINTER_WIDTH;
+                let address_of_bp = address_of_ip.checked_sub(POINTER_WIDTH)?;
                 // Can assume this resolves because we already walked over it when
                 // checking address_of_ip values.
-                let bp = stack_memory.get_memory_at_address(address_of_bp as u64)?;
+                let bp: u64 = stack_memory.get_memory_at_address(address_of_bp as u64)?;
```

Comment:
So keeping in mind that the address_of_bp calculation *isn't* necessarily already-walked over, this code needs to be completely reworked. 

I'm *very* dubious of entering this entire block when i==0. I'm always off-by-one with stackpointer stuff because everything goes backwards but it feels like we should never try to do this analysis in that case?

If we found the instruction pointer on *the first* iteration (or if we start doing misaligned searches, for any number of iterations that would cover less memory than an entire pointer), then that means the function pushed *nothing* to the stack (or less than a pointer). This by-defintion means there is no base-pointer to recover!

So I think the *correct* fix is to just this entire last_bp analaysis on `i > 0`? And then the comment above *would* be correct and the above subtraction would be immune to underflow.